scijava/scijava-ops-engine/src/main/java/org/scijava/ops/engine/yaml/YAMLJavaFieldInfoCreator.java

```diff
@@ -0,0 +1,51 @@
+
+package org.scijava.ops.engine.yaml;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.util.Map;
+
+import org.scijava.ops.api.OpInfo;
+import org.scijava.ops.engine.matcher.impl.OpFieldInfo;
+import org.scijava.ops.engine.matcher.impl.OpMethodInfo;
+
+public class YAMLJavaFieldInfoCreator implements YAMLOpInfoCreator {
+
+	@Override
+	public boolean canCreateFrom(String source, String identifier) {
+		return source.equals("Java") && identifier.indexOf('$') != -1;
+	}
+
+	@Override
+	public OpInfo create(final Map<String, Object> yaml, final String version) throws Exception {
+		final String[] names;
+		if (yaml.containsKey("name")) {
+			names = new String[] { (String) yaml.get("name") };
+		}
+		else {
+			String namesString = (String) yaml.get("names");
+			names = namesString.split("\\s*,\\s*");
+		}
+
+		double priority = 0.0;
+		if (yaml.containsKey("priority")) {
```

Comment:
Add an abstract superclass, implemented by the `Field`/`Method`/`Class` `InfoCreator`s, that handles `name` and `priority` and delegates the `source` parsing to the concrete implementation.